src/delay.rs

```diff
@@ -9,12 +9,14 @@ pub struct McycleDelay {
 
 impl McycleDelay {
     /// Constructs the delay provider
+    #[inline(always)]
     pub fn new(ticks_second: u32) -> Self {
         Self { ticks_second }
     }
 }
 
 impl DelayUs<u64> for McycleDelay {
+    #[inline(always)]
```

Comment:
I'd mark this as `#[inline]` since it's not a trivial function to inline it always.

Reply:
Yes, they are non trivial functions, marked as #[inline] instead.